docs/src/charm/howto/openstack.md

```diff
@@ -0,0 +1,120 @@
+# Integrating with OpenStack
+
+This guide explains how to integrate {{product}} with the OpenStack cloud
+platform.
+
+## Prerequisites
+
+To follow this guide, you will need:
+
+- An [OpenStack][openstack] cloud environment.
+- Juju set-up as the deployment tool for Openstack. Please refer to
+  [Juju and Openstack][juju-openstack] documentation for more information.
+- A Juju [controller][controller] with [access][credentials] to the OpenStack 
+  cloud environment.
```

Comment:
again -- this isn't super necessary.  If you have credentials to openstack, you can TOTALLY bootstrap a VM in openstack to use as your juju controller.  

Does any of this from charmed-kubernetes help?
https://ubuntu.com/kubernetes/docs/openstack-integration